### ISSUE TITLE
C++11 lambdas curly brackets are 1 indent off, missing option?

### ISSUE THREAD

**Issue by mlt** [uncrustify/uncrustify#410]
As of right now (uncrustify-0.61.3-gf65394e-win32), curly brackets for C++11 lambda are placed on the same indentation level as enclosing function call, i.e. making lambda's body match what otherwise would be line continuation. It would be nice to add an option to control whether general indentation rules shall apply for lambda body instead of line continuation rules.
Instead of

```
        m_ShutdownTimer.async_wait (
            [](const boost::system::error_code& ecode)
        {
            Daemon.running = 0;
        });
```

I'd like an option to have

```
        m_ShutdownTimer.async_wait (
            [](const boost::system::error_code& ecode)
            {
                Daemon.running = 0;
            });
```

Did I miss something? There are only 2 lambda options.
I have

```
indent_continue                           = 0        # number
indent_with_tabs                          = 1        # number
indent_func_call_param                    = true
sp_cpp_lambda_assign                      = ignore   # ignore/add/remove/force
sp_cpp_lambda_paren                       = remove   # ignore/add/remove/force
```

**Comment by gmaurel** [comment]
I take it in the TODO list.

**Comment by kankaristo** [comment]
I have a similar issue.

For example, this:
```cpp
    return std::shared_ptr<SDL_Surface>(
        image,
        [](SDL_Surface* surface) {
            LOG_DEBUG << "Freeing SDL_Surface: " << surface;
            SDL_FreeSurface(surface);
        }
    );
```

Turns into:
```cpp
    return std::shared_ptr<SDL_Surface>(
        image,
        [](SDL_Surface* surface) {
        LOG_DEBUG << "Freeing SDL_Surface: " << surface;
        SDL_FreeSurface(surface);
    }
);
```

Similarly, this:
```cpp
    auto sortListing = (
        [](const std::string &a, const std::string &b) {
            // Function body
        }
    );
```

Turns into:
```cpp
    auto sortListing = (
        [](const std::string &a, const std::string &b) {
        // Function body
    }
);
```

**Comment by exploser** [comment]
I just tried the new option and noticed that it only fixes indentation for lambdas that have parentheses in the declaration. When that is not the case, however, lambdas still get indented as if indent_cpp_lambda_body is false.

For example,
```
 auto foo = (
        []() {
            // Function body
        }
        );
```
remains fine, but if I use a shorthand version:
```
 auto foo = (
        [] {
            // Function body
        }
        );
```
the result is:
```
 auto foo = (
        [] {
            // Function body
    }
        );
````

(the ending parentheses are aligned with the content in [my current setup](https://github.com/uncrustify/uncrustify/files/638300/uncrustify.txt))

**Comment by gmaurel** [comment]
waiting for a PR from vyacheslavs

**Comment by guy-maurel** [comment]
Fixed at 529367ad790548d141a6ba82f155475582899a43
